### DISCUSSION TITLE
Error sending ROS 2 action goal: `The passed action type is invalid`

### DISCUSSION THREAD

**Discussion by switchdoclabs** [iRobotEducation/create3_docs#178] [Questions] [answered]
Hi,

I have installed and verified the ROS2 installation and the ability to talk to the Create3 Robot.

I type "ros2 action list -t" and I get the actions for the Create 3 robot.

When I try to execute the action /dock I get "the passed action type is invalid"

create3@create3-desktop:$ ros2 action send_goal /dock irobot_create_msgs/action/DockServo "{}"
The passed action type is invalid
create3@create3-desktop:$ ros2 action list -t
/audio_note_sequence [irobot_create_msgs/action/AudioNoteSequence]
/dock [irobot_create_msgs/action/DockServo]
/drive_arc [irobot_create_msgs/action/DriveArc]
/drive_distance [irobot_create_msgs/action/DriveDistance]
/led_animation [irobot_create_msgs/action/LedAnimation]
/navigate_to_position [irobot_create_msgs/action/NavigateToPosition]
/rotate_angle [irobot_create_msgs/action/RotateAngle]
/undock [irobot_create_msgs/action/Undock]
/wall_follow [irobot_create_msgs/action/WallFollow]
create3@create3-desktop:~$

What am I doing wrong?

John

**Comment by justinIRBT** [comment, marked as answer]
Sounds like your machine is missing irobot_create_msgs in its ros2 environment.  If you are running an ubuntu machine with ros2 apt installed, did you install
ros-galactic-irobot-create-msgs
otherwise you can clone the repo https://github.com/iRobotEducation/irobot_create_msgs and build it with your ros2 sandbox

**Reply by switchdoclabs** [reply]
Thanks for your response.    I have installed irobot_create_messages in the environment.

ros2 action list -t shows the actions.   But ros2 action doesn't see them.  Is there something else I need to do?

create3@create3-desktop:$ ros2 action list -t
/audio_note_sequence [irobot_create_msgs/action/AudioNoteSequence]
/dock [irobot_create_msgs/action/DockServo]
/drive_arc [irobot_create_msgs/action/DriveArc]
/drive_distance [irobot_create_msgs/action/DriveDistance]
/led_animation [irobot_create_msgs/action/LedAnimation]
/navigate_to_position [irobot_create_msgs/action/NavigateToPosition]
/rotate_angle [irobot_create_msgs/action/RotateAngle]
/undock [irobot_create_msgs/action/Undock]
/wall_follow [irobot_create_msgs/action/WallFollow]

**Reply by justinIRBT** [reply]
What you are describing I believe is the behavior if the messages aren’t in the sourced ROS environment.  “action list -t” will still show the message types in that case, but the system can’t use them.

Did you install from the apt package or in a development sandbox?  If you installed from the apt package, can you inspect the /opt/ros/galactic/ directory to see if you can find irobot_create_msgs.  If you built them from a development sandbox, make sure you are sourcing the setup.bash from the install directory in the development sandbox.  

To confirm it is an issue with not having the message definitions in the environment, try the same thing with topics.  “topic list -t” should show topics, but “topic echo” will fail.

**Reply by switchdoclabs** [reply]
Thank you for your help!   I had not put source install/setup.bash in my file.

That fixed it.   I was suspecting something like that but you saved me a lot of debugging.

I just ordered 8 more of Create3 for my robotics students!  I had to get it to work before I handed it over.  :)

John

**Comment by swilloughbypme** [comment]
I was having the same problem as the OP, but I had updated my Create 3 to the current Humble firmware (H.0.0). This thread solved the problem except (and maybe this is obvious):

Rather than installing this package...
- ros-**galactic**-irobot-create-msgs
Use this one:
- ros-**humble**-irobot-create-msgs

I've now entered the Happy Camper state. For other info, see the Humble branch in the msgs repo.

Steve
